.github/workflows/pytest.yaml

```diff
@@ -8,32 +8,41 @@ on:
       - master
   release:
     types: [published]
-    
+
 jobs:
-  tests:
+  test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      
-      - name: Install Python 2.7
-        uses: actions/setup-python@v3
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Build and Run Tests in Docker
+        uses: docker/build-push-action@v2
         with:
-          python-version: 2.7
-      
-      - name: Install dependencies
-        run: |
-          python --version
-          python -m pip install --upgrade pip setuptools wheel
-          pip install --no-cache-dir typing==3.7.4.1
-          pip install --ignore-installed -r requirements.txt -e .[tests,docs]
-      
+          context: .
+          dockerfile: Dockerfile
+          tags: tests_image
+          build-args: |
+            PYTHON_VERSION=2.7
+
       - name: Run tests with pytest and generate report
-        run: coverage run -m pytest
-      
+        run: >
+          docker run
+          --name tests_container
+          --entrypoint pytest
+          --volume $(pwd):/app
+          tests_image
+          tests
+          --cov=./
+          --cov-report=xml
+
       - name: Upload Coverage to Codecov
         uses: codecov/codecov-action@v2
-      
+
       - name: Build and push
+        if: ${{ github.event.pull_request.merged }}
```

Comment:
you can use the same as https://github.com/SCOAP3/hepcrawl/pull/109/files#diff-038d392b404f94b2ee8e3f10b6ae76784160379d9e90638f78ee65df63353221R55 :)